test/server/parser/parser.ts

```diff
@@ -61,15 +90,332 @@ function getSyntaxFiles(): string[]
 	return readdirSync(`${__dirname}/../../${syntaxDir}/`).filter(file => !file.startsWith('fixme_'))
 }
 
-test('Grove files with no syntax errors', () =>
+describe('Basic syntax tests', () =>
 {
-	const error = jest.spyOn(global.console, 'error')
+	test('Grove files with no syntax errors', () =>
+	{
+		const error = jest.spyOn(global.console, 'error')
 
-	const syntaxFiles = getSyntaxFiles()
-	for (const file of syntaxFiles)
-		expect(file).toBeValidSyntax()
+		const syntaxFiles = getSyntaxFiles()
+		for (const file of syntaxFiles)
+			expect(file).toBeValidSyntax()
 
-	// TODO: print file name if spy detects or `console.error`
-	expect(error).not.toHaveBeenCalled()
+		// TODO: print file name if spy detects or `console.error`
+		expect(error).not.toHaveBeenCalled()
+	})
 })
 
+describe('AST Nodes tests', () =>
+{
+	test('generic_function_call_comparison_simple.grove parsing', () =>
+	{
+		// Parse the source file
+		const parser = parserFor('generic_funciton_call_comparison_simple.grove')
```

Comment:
`funciton` -> `function`?

We're guessing this file doesn't actually stick around, as if it did we'd ask it to use lowerCamelCase naming unless we want to break from style here to make test-generated file names stand out?

Reply:
I think the `grove` file names should follow the official mangrove coding style. But since it doesn't, my rust brain went with snake case. I can rename them using lowerCamelCase

Reply:
That would be grand, please, as Mangrove's a lowerCamelCase language (it's what makes us most comfy as _'d names don't read for us as-if they had spaces in.. brains are weird huh?).